File: Python/001 Two Sum.py
# -*-coding:utf-8 -*-

class Solution:
    def twoSum(self, nums, target):
        """
        :type nums: List[int]
        :type target: int
        :rtype: List[int]
        """
        counter = 0 
        for i in nums:
            counter += 1
            another_num = target - i
            if another_num in nums[counter:]:
                return [counter - 1, nums[counter:].index(another_num) + counter]


s =Solution()
print s.twoSum([2,5,5,11],10)

File: JavaScript/001 Two Sum.js
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number[]}
 */
var twoSum = function(nums, target) {
    for (var i = 0; i < nums.length; i++) {
        var j = target - nums[i];
        var surplus = nums.slice(i + 1);
        if (surplus.indexOf(j) !== -1) {
            return [i, surplus.indexOf(j) + 1 + i];
        }
    }
};

console.log(twoSum([2, 5, 5, 11], 10));